### DISCUSSION TITLE
Potion Upgrade

### DISCUSSION THREAD

**Discussion by RGFTheCoder** [P3pp3rF1y/SophisticatedBackpacks#146] [Ideas]
An upgrade that would work similarly to the feeding upgrade and would automatically drink potions when you didn't have their effects anymore;

By default, if there is a potion with multiple effects, it should use it if any of the effects are missing, but as a last resort (use single-effect potions first). 

An advanced version could allow you to choose what effects you want enabled, rather than all of them at once.

**Comment by TheDeviantCrafter** [comment]
To keep the UI simple, I would give it the same grid-based filter that the other upgrades have. When multiple potion are present that can give the same potion effect, it could choose the top-left most one. If the highest priority potion has an effect that would be wasted it will attempt to drink a lower priority potion that doesn't have the wasted potion effect.
